airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -108,6 +108,24 @@ def _parse_date_value(value) -> datetime:
     return record_value >= state_value
 
 
+def is_per_stream_state(message: AirbyteMessage) -> bool:
+    return message.state and isinstance(message.state, AirbyteStateMessage) and message.state.type == AirbyteStateType.STREAM
+
+
+def construct_latest_state_from_messages(messages: List[AirbyteMessage]) -> Dict[str, Mapping[str, Any]]:
+    """
+    Because connectors that have migrated to per-stream state only emit state messages with the new state value for a single
+    stream, this helper method reconstructs the final state of all streams after going through each AirbyteMessage
+    """
+    latest_per_stream_by_name = dict()
+    for message in messages:
+        current_state = message.state
+        if current_state and current_state.type == AirbyteStateType.STREAM and hasattr(current_state, "stream"):
```

Comment:
out of curiosity why not use `current_state.stream` instead of `hasattr`?

Reply:
I think that is this statement `hasattr(current_state, "stream")` is always truthy, because we first check that the state type is `STREAM.